### PR TITLE
Support quant output

### PR DESCRIPTION
**Pull request by asenina**

### CODE REVIEW

File: nncf/quantization/quantizer_propagation.py [hunk 1]
```diff
@@ -1687,15 +1688,18 @@ def _setup_initial_quantizers_for_operator_node(self, operator_node_key: str,
...
             return  # TODO: remove this once module insertion points are included in the IP graph
 
+        if not self._quantize_outputs and MODEL_OUTPUT_OP_NAME in operator_node_key:
+            return
         # No need to place quantizers for FP32-forced ops, naturally
         if node[QuantizerPropagationStateGraph.QUANTIZATION_TRAIT_NODE_ATTR] in \
                 [QuantizationTrait.NON_QUANTIZABLE,
                  QuantizationTrait.QUANTIZATION_AGNOSTIC,
-                 QuantizationTrait.CONCAT]:
+                 QuantizationTrait.CONCAT] and MODEL_OUTPUT_OP_NAME not in operator_node_key:
```
Comment (vshampor) [inline nncf/quantization/quantizer_propagation.py:1697]: Seems to be redundant with respect to the condition above.

**Comment by vshampor** [comment]
Just in case: tests are missing, especially the ones that test integration with `scope_overrides`.

File: tests/data/reference_graphs/original/gnmt_variable.dot [hunk 2]
```diff
@@ -1,389 +0,0 @@
...
-"166 ResidualRecurrentDecoder/ModuleList[rnn_layers]/NNCF_RNN[2]/StackedRNN[rnn_impl]/StackedRNNResetPoint/cat" -> "169 ResidualRecurrentDecoder/__add__";
-"169 ResidualRecurrentDecoder/__add__" -> "170 ResidualRecurrentDecoder/Classifier[classifier]/Linear[classifier]/linear";
-"171 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/__add__" -> "172 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/tanh";
-"172 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/tanh" -> "173 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/matmul";
-"173 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/matmul" -> "174 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/softmax";
-"174 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/softmax" -> "175 ResidualRecurrentDecoder/RecurrentAttention[att_rnn]/BahdanauAttention[attn]/bmm";
-}
```
Comment (vshampor) [inline tests/data/reference_graphs/original/gnmt_variable.dot:389]: Why did this file have to be removed?

File: tests/test_compressed_graph.py [hunk 3]
```diff
@@ -752,7 +752,7 @@ def forward(self, x):
 
 def test_output_quantization(_case_config):
     # TODO: Add support "quantize_outputs" option in propagation mode.
-    pytest.skip()
+    #pytest.skip()
```
Comment (vshampor) [inline tests/test_compressed_graph.py:755]: No commented code allowed.

Comment (asenina) [reply tests/test_compressed_graph.py]: Done.

File: tests/data/reference_graphs/original/gnmt_variable.dot (hunk 2 above)
Comment (asenina) [reply tests/data/reference_graphs/original/gnmt_variable.dot:389]: This is an extra file. It doesn't match any test case.

File: tests/quantization/test_algo_quantization.py [hunk 4]
```diff
@@ -448,3 +449,63 @@ def test_quantizer_ordering(requanting_qconf: QuantizerConfig,
...
+    config['target_device'] = "TRIAL"
+    config['compression']['scope_overrides'] = {
+        "nncf_model_output_0": {
+            "bits": 4,
+            "mode": "asymmetric",
+        }
+    }
+    model, _ = create_compressed_model_and_algo_for_test(model, config)
+    output_quantizers =\
+        [quantizer for scope_str, quantizer in model.activation_quantizers.items() if "conv2d" in scope_str]
```
Comment (vshampor) [inline tests/quantization/test_algo_quantization.py:508]: Use controller attributes for this kind of access to information on quantizer modules..

File: tests/test_nncf_network.py [hunk 5]
```diff
@@ -754,7 +755,7 @@ def test_multiple_forward():
     config = get_basic_sparsity_plus_quantization_config()
     sparse_quantized_model, _ = create_compressed_model_and_algo_for_test(model, config)
     graph = sparse_quantized_model.get_original_graph()
-    for node_key in list(graph.get_all_node_keys())[1:]:
+    for node_key in list(graph.get_all_node_keys())[1:4]:
```
Comment (vshampor) [inline tests/test_nncf_network.py:758]: `[1:-1]`?

**Review by vshampor — CHANGES_REQUESTED** [review]

**Review by alexsu52 — CHANGES_REQUESTED** [review]
The major request that I see, that the inputs and outputs nodes should not be identifier as nodes with in_degree=0 and out_degree=0 but they should be specified explicitly. I think It can be implemented in the common part.

File: tests/quantization/test_algo_quantization.py (hunk 4 above)
Comment (asenina) [reply tests/quantization/test_algo_quantization.py]: Done.

File: tests/test_nncf_network.py (hunk 5 above)
Comment (asenina) [reply tests/test_nncf_network.py]: Done.

File: nncf/common/graph/graph.py [hunk 6]
```diff
@@ -91,6 +91,9 @@ def get_input_nodes(self) -> List[NNCFNode]:
                 inputs.append(self._nx_node_to_nncf_node(self._nx_graph.nodes[nx_node_key]))
         return inputs
 
+    def get_output_nodes(self):
+        return self.get_graph_outputs()
```
[the remark this answers is not in the document]
Comment (asenina) [reply nncf/common/graph/graph.py]: Implementation moved to `get_output_nodes`. Removed `get_graph_outputs`.
 Done.

File: nncf/common/pruning/model_analysis.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
 from nncf.common.pruning.utils import PruningOperationsMetatypeRegistry
 
 
+
```
Comment (alexsu52) [inline nncf/common/pruning/model_analysis.py:23]: Please, remove the extra blank line.

File: nncf/common/graph/graph.py [hunk 8]
```diff
@@ -91,7 +91,7 @@ def get_input_nodes(self) -> List[NNCFNode]:
                 inputs.append(self._nx_node_to_nncf_node(self._nx_graph.nodes[nx_node_key]))
         return inputs
 
-    def get_graph_outputs(self) -> List[NNCFNode]:
+    def get_output_nodes(self):
```
Comment (alexsu52) [inline nncf/common/graph/graph.py:94]:
```suggestion
    def get_output_nodes(self) -> List[NNCFNode]:
```

File: nncf/common/pruning/model_analysis.py [hunk 9]
```diff
@@ -333,3 +335,9 @@ def analyse_model_before_pruning(self):
         self.propagate_can_prune_attr_up()
         self.propagate_can_prune_attr_down()
         return self.can_prune
+
+    def _ignore_nncf_output_nodes(self, list_nncf_nodes):
```
[the remark this answers is not in the document]
Comment (asenina) [reply nncf/common/pruning/model_analysis.py]: It's removed. Added `PTInput` and `PTOutput`  for correctly process input/output nodes.
Done.

**Review by alexsu52 — APPROVED** [review]

File: nncf/common/pruning/model_analysis.py (hunk 7 above)
Comment (asenina) [reply nncf/common/pruning/model_analysis.py]: Done.

File: nncf/common/graph/graph.py (hunk 8 above)
Comment (asenina) [reply nncf/common/graph/graph.py]: Done.

**Review by ljaljushkin — APPROVED** [review]

**Review by AlexanderDokuchaev — APPROVED** [review]

**Review by vshampor — APPROVED** [review]

**Pull request merged**